x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/physical/local/InsertFieldExtraction.java

```diff
@@ -102,15 +101,18 @@ public PhysicalPlan apply(PhysicalPlan plan) {
 
     private static Set<Attribute> missingAttributes(PhysicalPlan p) {
         var missing = new LinkedHashSet<Attribute>();
-        var inputSet = p.inputSet();
+        var input = new AttributeSet(p.inputSet());
 
-        // TODO: We need to extract whatever fields are missing from the left hand side.
-        // skip the lookup join since the right side is always materialized and a projection
+        // For LOOKUP JOIN we only need field-extraction on left fields used to match, since the right side is always materialized
```

Comment:
The field extraction is implicitly done inside the `LookupFromIndexService`. I agree it'd be nicer to make the field extraction visible in the physical plan; currently, that'd have no benefit, though, because we cannot really materialize fields late - all looked up fields need to be extracted inside the `LookupFromIndexService`.

It'd require a fundamental change of the execution to extract additional fields _after_ the execution of the lookup join.